core/src/main/java/org/elasticsearch/index/translog/ImmutableTranslogReader.java

```diff
@@ -88,4 +158,20 @@ public Checkpoint getInfo() {
         return new Checkpoint(length, totalOperations, getGeneration());
     }
 
+    @Override
+    public final void close() throws IOException {
+        if (closed.compareAndSet(false, true)) {
+            channel.close();
+        }
+    }
+
+    protected final boolean isClosed() {
+        return closed.get();
+    }
+
+    protected void ensureOpen() {
+        if (isClosed()) {
+            throw new AlreadyClosedException("translog [" + getGeneration() + "] is already closed");
```

Comment:
Can we change `getGeneration()` to `toString()` instead so the exception also tells us which file was problematic?